sycl/include/sycl/handler.hpp

```diff
@@ -1548,7 +1564,7 @@ class __SYCL_EXPORT handler {
     //       conflicts they should be included in the name.
     using NameT =
         typename detail::get_kernel_name_t<KernelName, KernelType>::name;
-    verifyUsedKernelBundle(detail::KernelInfo<NameT>::getName());
+    verifyUsedKernelBundle(detail::getKernelName<NameT>());
```

Comment:
nit: We cannot pass `NameT` accross ABI boundary but we can use a tiny helper

```
template <typename KernelNameTy>
verifyUsedKernelBundle() {
  verifyUsedKernelBundle(detail::getKernelName<KernelNameTy>());
}
```
to clean up the callsites a bit.